app/models/factory.rb

```diff
@@ -0,0 +1,7 @@
+class Factory < ActiveRecord::Base
+  belongs_to :manufacturer
+
+  validates :city, uniqueness: true, presence: true
+  validates :state, uniqueness: true, presence: true
```

Comment:
likewise don't validate uniqueness for state, even scoped to manufacturer (bc a manufacturer can have multiple factories in the same state)
to ensure the state attribute is a real US state, 
validates :state, inclusion: { in: %w(AL AK AR....)} (and you'd have to write out the whole array)